### ISSUE TITLE
Student submitting fewer responses than allowed: don't consider the question as 'not fully answered'

### ISSUE THREAD

**Issue by damithc** [TEAMMATES/teammates#11992]
v8.19.2

Background: an questions is set to allow a max number of responses:
![image](https://user-images.githubusercontent.com/1673303/200155822-b5a0d50b-598a-46e5-8306-4a5474a19d3a.png)

Problem:
If a student give fewer than 10 answers to the above question, the question is counted as 'not fully answered' and a warning is shown. This could be misleading to the user, as the _max response count allowed_ is a restriction rather than a requirement.

![image](https://user-images.githubusercontent.com/1673303/200156031-c4407a9e-7df7-4d82-acef-0054433fe40f.png)

Suggestion:
Consider the question as 'answered' even if the number of the responses is fewer than the max allowed
What do others think?

**Comment by jwang444** [comment]
Hey, can we claim this issue for class project (University of Michigan)?

**Comment by fsgmhoward** [comment]
@jwang444 Sure. Everyone is welcomed to claim it since no one is doing so now. Please feel free to submit a PR for this.

**Comment by cheweejia** [comment]
Hi @damithc, I agree with your suggestion. Perhaps the phrasing of the success message can be modified to indicate that the questions are **unanswered** instead of the fact that they are **yet to be fully answered**. The difference to me is that the former suggests the student's deliberate intention to leave the response empty, while the latter seems to imply that the student has yet to complete their response (hence it can be misleading).

An example of the change may be:
"!Note that the responses to the following questions have been left empty by you. They are: 2."

Not sure if @jwang444 is still working on this since there is no PR created yet, but feel free to offer your thoughts/take up the suggestion or let me know if you do not intend to work on it anymore. Thank you!

**Comment by fsgmhoward** [comment]
@cheweejia Since @jwang444 has not responded for a period of time. You may proceed to work on the issue and feel free to submit an PR. Thanks!

**Comment by jwang444** [comment]
Hello!

Sorry for the late response. Me and my partner are still working on these and will make PRs soon.

**Comment by wangdan87** [comment]
After investigating the issue a little bit, it seems that the wording could be something like "Note that you have not submitted the maximum possible number of responses for the following questions: 2." This would make sense because the way I understand the issue, when a user gives feedback to their teammates for example, they don't have to give feedback to everyone on their team and can just choose a few teammates to evaluate.

**Comment by damithc** [comment]
> After investigating the issue a little bit, it seems that the wording could be something like "Note that you have not submitted the maximum number of responses allowed for the following questions: 2." This would make sense because the way I understand this feature, when a user gives feedback to their teammates for example, they don't have to give feedback to everyone on their team and can just choose a few teammates to evaluate. The new wording would give the user a warning that they could still add responses to the specified question without making it seem like the question is unanswered.

@wangdan87 Yes, that can work. The simpler solution is to consider them as 'fully answered' as long as there is at least the minimum number of responses required.

**Comment by kevin9foong** [comment]
Could I give this issue an attempt?

**Comment by samuelfangjw** [comment]
> Could I give this issue an attempt?

@kevin9foong You're welcome to try!

**Comment by Abhijeetmishr** [comment]
> > Could I give this issue an attempt?
> 
> @kevin9foong You're welcome to try!

@samuelfangjw is this issue is fixed or I can give it a try ?

**Comment by zhaojj2209** [comment]
@Abhijeetmishr Feel free to submit a PR for this issue!

**Comment by Abhijeetmishr** [comment]
> @Abhijeetmishr Feel free to submit a PR for this issue!

@zhaojj2209 Please assign this to me
Regards

**Comment by zhaojj2209** [comment]
@Abhijeetmishr As per our [development guidelines](https://teammates.github.io/teammates/process.html#step-1-pick-an-issue-to-fix), there is no need for contributors to be assigned issues.

**Comment by Abhijeetmishr** [comment]
> @Abhijeetmishr As per our [development guidelines](https://teammates.github.io/teammates/process.html#step-1-pick-an-issue-to-fix), there is no need for contributors to be assigned issues.

Cool

**Comment by Abhijeetmishr** [comment]
@zhaojj2209 changing this below comment will resolve this issue right 
`  <i class="fas fa-exclamation"></i> **Note that some questions are yet to be fully answered.
    They are: {{ notYetAnsweredQuestions.join(', ')** }}.`

**Comment by zhaojj2209** [comment]
@Abhijeetmishr The fix is not as simple as simply changing the wording of the warning. Please refer to the above discussion for suggestions on how to solve the issue.

**Comment by Abhijeetmishr** [comment]
> @Abhijeetmishr The fix is not as simple as changing the wording of the warning. Please refer to the above discussion for suggestions on how to solve the issue.

@zhaojj2209 It means we have to make change:-
1. in Backend 
2. as well as FE
I am not able to understand above requirement that 
1. do we have to remove warning with some meaningful message
2. or we have to remove the max no of answers for particular question.

correct me if I am wrong and Please provide set of instruction which I can follow to get this task done.

Regards

**Comment by zhaojj2209** [comment]
@Abhijeetmishr There have been two solutions proposed for this issue:

1. Consider questions with fewer responses than the maximum allowed responses as "fully answered" so that the "not fully answered" warning does not show up for these questions.
2. Instead of showing the "not fully answered" warning, show a different warning for questions with fewer responses than the maximum allowed responses.

Both solutions are acceptable, I would personally recommend the first solution.

Hope the above information is enough to get you started. We will not provide an instruction manual for you; please figure out the rest on your own and we will review your approach and make comments should you submit a PR.

**Comment by Abhijeetmishr** [comment]
@zhaojj2209 how do I reproduce the above issue using teammates app please guide.

**Comment by Abhijeetmishr** [comment]
@damithc @zhaojj2209 **`<p id="not-answered-questions" *ngIf="notYetAnsweredQuestions.length === 10">
    <i class="fas fa-exclamation"></i> Note that some questions need to be answered.
  </p>`**
  this will work if I add **else condtion with this and with message "fully answered."**

**Comment by zhaojj2209** [comment]
> this will work if I add **else condtion with this and with message "fully answered."**

No. A question is considered answered even if they do not hit the max number of responses.

**Comment by domlimm** [comment]
@MMilou24 Hello again! As this is your first time contributing to TEAMMATES, would you like to look at an issue tagged with `good first issue` as your first? If not, by all means you may go for this! 👍🏻
